### PR TITLE
Remove remaining old-style AcceleratorConnector properties

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Removes two remaining properties on the AcceleratorConnector that are unused. 
We voted for removing them directly without deprecation due to the connector being an internal component.

### Does your PR introduce any breaking changes? If yes, please list them.

Yes but it's very minor. Connectors are internal and are not meant to be accessed by users.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/Lightning-AI/lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @justusschock @awaelchli @rohitgr7 @tchaton @kaushikb11 @ninginthecloud @akihironitta

### CODE REVIEW

File: tests/tests_pytorch/deprecated_api/test_remove_1-8.py [hunk 1]
```diff
@@ -1132,17 +1132,37 @@ def test_trainer_gpus(monkeypatch, trainer_kwargs):
...
     ):
         assert trainer.gpus == trainer_kwargs["devices"]
 
+    with pytest.deprecated_call(
+        match=(
+            "`AcceleratorConnector.gpus` was deprecated in v1.6 and will be removed in v1.8. Please use"
+            " `Trainer.num_devices` or `Trainer.device_ids` to get device information instead."
+        )
+    ):
+        assert trainer._accelerator_connector.gpus == trainer_kwargs["devices"]
```
Comment (rohitgr7) [inline tests/tests_pytorch/deprecated_api/test_remove_1-8.py:1148]: since accelerator_connector is protected.. should it rather be removed?

**Review by justusschock — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: tests/tests_pytorch/deprecated_api/test_remove_1-8.py (hunk 1 above)
Comment (otaj) [reply tests/tests_pytorch/deprecated_api/test_remove_1-8.py:1148]: I'd be a bit cautious about removing it completely. Sure, it's protected, so officially "no guarantees", but it's not like it's hard to find and I can easily imagine some user code using it

**Review by otaj — APPROVED** [review]

File: tests/tests_pytorch/deprecated_api/test_remove_1-8.py (hunk 1 above)
Comment (rohitgr7) [reply tests/tests_pytorch/deprecated_api/test_remove_1-8.py:1148]: we provide separate trainer properties for them.

Comment (awaelchli) [reply tests/tests_pytorch/deprecated_api/test_remove_1-8.py]: Fine then let's just remove them

**Review by rohitgr7 — APPROVED** [review]

**Pull request merged**